rb_ws/src/buggy/scripts/auton/autonsystem.py

```diff
@@ -1,187 +1,281 @@
         self.steer_publisher = rospy.Publisher(
-            buggy_name + "/input/steering", Float64, queue_size=1
+            self_name + "/input/steering", Float64, queue_size=1
         )
         self.brake_publisher = rospy.Publisher(
-            buggy_name + "/input/brake", Float64, queue_size=1
+            self_name + "/input/brake", Float64, queue_size=1
         )
         self.brake_debug_publisher = rospy.Publisher(
-            buggy_name + "/auton/debug/brake", Float64, queue_size=1
+            self_name + "/auton/debug/brake", Float64, queue_size=1
         )
         self.heading_publisher = rospy.Publisher(
-             buggy_name + "/auton/debug/heading", Float32, queue_size=1
+            self_name + "/auton/debug/heading", Float32, queue_size=1
         )
         self.distance_publisher = rospy.Publisher(
-             buggy_name + "/auton/debug/distance", Float64, queue_size=1
+            self_name + "/auton/debug/distance", Float64, queue_size=1
         )
 
+
         self.auton_rate = 100
         self.rosrate = rospy.Rate(self.auton_rate)
 
+        self.profile = profile
         self.tick_caller()
 
-    def update_speed(self, msg):
+    def update_self_odom(self, msg):
+        with self.lock:
+            self.self_odom_msg = msg
+
+    def update_other_odom(self, msg):
         with self.lock:
-            self.speed = msg.data
+            self.other_odom_msg = msg
 
-    def update_msg(self, msg):
+    def update_other_steering_angle(self, msg):
         with self.lock:
-            self.msg = msg
-        
+            self.other_steering = msg.data
+
     def tick_caller(self):
-        while ((not rospy.is_shutdown()) and (self.msg == None)): # with no message, we wait
+        while ((not rospy.is_shutdown()) and
+            (self.self_odom_msg == None or
+            (self.has_other_buggy and self.other_odom_msg == None))): # with no message, we wait
             rospy.sleep(0.001)
-        
+
         # wait for covariance matrix to be better
         while ((not rospy.is_shutdown()) and
-               (self.msg.pose.covariance[0] ** 2 + self.msg.pose.covariance[7] ** 2 > 1**2)):
+               (self.self_odom_msg.pose.covariance[0] ** 2 + self.self_odom_msg.pose.covariance[7] ** 2 > 1**2)):
             # Covariance larger than one meter. We definitely can't trust the pose
             rospy.sleep(0.001)
             print("Waiting for Covariance to be better: ",  rospy.get_rostime())
         print("done checking covariance")
-        while (not rospy.is_shutdown()): # start the actual control loop
-            self.tick()
+
+        # initialize global trajectory index
+
+        with self.lock:
+            e, _ = self.get_world_pose_and_speed(self.self_odom_msg)
+
+        while (not rospy.is_shutdown()):
+            # start the actual control loop
+            # run the planner every 10 ticks
+            # thr main cycle runs at 100hz, the planner runs at 10hz, but looks 1 second ahead
```

Comment:
"the"

and it looks 2s ahead according to the path planning code. Perhaps just reference the constant variable in case we change the projection time.

Reply:
change added